astropy/tests/helper.py

```diff
@@ -604,3 +604,40 @@ def check_pickling_recovery(original, protocol):
     class_history = [original.__class__]
     generic_recursive_equality_test(original, unpickled,
                                     class_history)
+
+
+def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=0, err_msg='', verbose=True):
+    """
+    Raise an assertion if two objects are not equal up to desired tolerance.
+
+    This is a :class:`~astropy.units.Quantity`-aware version of
+    :func:`numpy.testing.assert_allclose`.
+    """
+
+    import numpy as np
+    from ..units import Quantity
+
+    if isinstance(actual, Quantity) and isinstance(desired, Quantity):
```

Comment:
I agree with @mhvk that  this may be a call to try to get more explicit error messages in Quantity itself... But I think there's no harm in leaving this in to make reading of test results easier.